packages/rps/src/utils/channelClient.ts

```diff
@@ -0,0 +1,219 @@
+import {BigNumberish} from 'ethers/utils';
+import {EventEmitter} from 'events';
+
+export enum ChannelStatus {
+  Opening = 'opening',
+}
+
+export enum MethodName {
+  CreateChannel = 'CreateChannel',
+  JoinChannel = 'JoinChannel',
+  UpdateChannel = 'UpdateChannel',
+}
+
+export enum NotificationName {
+  ChannelProposed = 'ChannelProposed',
+  ChannelUpdated = 'ChannelUpdated',
+}
+
+export type ActionName = MethodName | NotificationName;
+
+// export type MessageParameters = OpenChannelParameters | JoinChannelParameters;
+
+export interface Participant {
+  /**
+   * App allocated id, used for relaying messages to the participant
+   */
+  participantId: string;
+
+  /**
+   * Address used to sign channel updates
+   */
+  signingAddress: string;
+
+  /**
+   * Address of EOA to receive channel proceeds (the account that'll get the funds).
+   */
+  destination: string;
+}
+
+export interface AllocationItem {
+  /**
+   * Address of EOA to receive channel proceeds.
+   */
+  destination: string;
+
+  /**
+   * How much funds will be transferred to the destination address.
+   */
+  amount: BigNumberish;
+}
+
+export interface Allocation {
+  /**
+   * The token's contract address.
+   */
+  token: string;
+
+  /**
+   * A list of allocations (how much funds will each destination address get).
+   */
+  allocationItems: AllocationItem[];
+}
+
+export interface Message {
+  /**
+   * Identifier or user that the message should be relayed to
+   */
+  participantId: string;
+
+  /**
+   * Message payload. Format defined by wallet and opaque to app.
+   */
+  data: string;
+}
+
+export interface Funding {
+  token: string;
+  amount: string;
+}
+
+export interface ParameterHash {
+  [key: string]: any;
+}
+
+export interface JsonRPCNotification<ParametersType = ParameterHash, MethodType = string> {
+  jsonrpc: '2.0';
+  method: MethodType;
+  params: ParametersType;
+}
+
+export interface JsonRPCRequest<ParametersType = ParameterHash, MethodType = string>
+  extends JsonRPCNotification<ParametersType, MethodType> {
+  id: string;
+}
+
+export interface JsonRPCResponse<ResultType = {[key: string]: any}> {
+  jsonrpc: '2.0';
+  id: string;
+  result: ResultType;
+}
+
+export interface CreateChannelParameters extends UpdateChannelParameters {
+  appDefinition: string;
+}
+
+export interface ChannelResult extends CreateChannelParameters {
+  channelId: string;
+  status: ChannelStatus;
+  funding: Funding[];
+}
+
+export interface CreateChannelRequest
+  extends JsonRPCRequest<CreateChannelParameters, MethodName.CreateChannel> {}
+
+export interface CreateChannelResult extends ChannelResult {
+  turnNum: 0;

```

Comment:
I don't think it's important to enforce this at the typing level. `turnNum` can just be a bigNum.